reference/concepts.md

```diff
@@ -0,0 +1,105 @@
+# Haskell reference
+
+## Concepts
+
+The Haskell concept exercises are based on concepts. The list below contains the concepts that have been identified for the Haskell language.
+
+### Language-unique
+
+- ~~Active patterns~~ Pattern synonyms? That seems a stretch.
+- ~~Computation expressions~~ do-notation
+- Dependency order
```

Comment:
could I hear what you imagine this concept will cover and/or a quick example?

Reply:
I also think it doesn't